compiler/ccgstmts.nim

```diff
@@ -834,8 +834,12 @@ proc genTryCpp(p: BProc, t: PNode, d: var TLoc) =
       endBlock(p)
     else:
       for j in 0..t[i].len-2:
-        assert(t[i][j].kind == nkType)
-        startBlock(p, "catch ($1*) {$n", getTypeDesc(p.module, t[i][j].typ))
+        if t[i][j].isInfixAs():
+          let v = rope($t[i][j][2])
```

Comment:
wait ... what? Pretty sure that's not how to generate C++ code.

Reply:
Agree, fixed in the last version